src/test/java/com/thealgorithms/misc/SparsityTest.java

```diff
@@ -0,0 +1,66 @@
+package com.thealgorithms.misc;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+public class SparsityTest {
+
+    private static final double DELTA = 1e-9;
+
+    @Test
+    public void testAllZeroElements() {
+        double[][] mat = {{0, 0, 0}, {0, 0, 0}, {0, 0, 0}};
```

Comment:
Could you please refactor these tests to parametrized, to avoid code duplication

Reply:
Refactored the test cases to Parameterized Tests!